library/core/src/macros/mod.rs

```diff
@@ -1287,13 +1287,20 @@ pub(crate) mod builtin {
 
     /// Expands to the file name in which it was invoked.
     ///
-    /// With [`line!`] and [`column!`], these macros provide debugging information for
-    /// developers about the location within the source.
-    ///
-    /// The expanded expression has type `&'static str`, and the returned file
-    /// is not the invocation of the `file!` macro itself, but rather the
-    /// first macro invocation leading up to the invocation of the `file!`
-    /// macro.
+    /// With [`line!`] and [`column!`], these macros provide debugging information for developers
+    /// about the location within the source.
+    ///
+    /// The expanded expression has type `&'static str`, and the returned file is not the invocation
+    /// of the `file!` macro itself, but rather the first macro invocation leading up to the
+    /// invocation of the `file!` macro.
```

Comment:
I am just a daily rust user, not a compiler expert but I have used file macro before and I think this sentence is relatively hard to grasp. I think an example could help.

Reply:
I understood it as follows: If `file!()` is called from another macro, then it returns the name of the file where the outer macro is called, not the file where the literal `file!()` invocation is.

If that's correct, the original description is clear to me.

Reply:
Sorry if it looked like I was implying the documentation was wrong. That was not my intention, I was just asking for an example because it usually helps to understand things. If y'all think it's clear enough, feel free to resolve that thread.

Reply:
While it can be useful to find ways to improve documentation, this is out of scope of this PR.  This was only called out because I didn't pay enough attention when I re-wrapped the lines and my editor and the author of this section's disagreed on the line wraps.  I have since reverted that change.